### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by cbruno10**

### CODE REVIEW

File: hibp/table_password.go [hunk 1]
```diff
@@ -2,125 +2,70 @@ package hibp
...
 func tablePassword() *plugin.Table {
 	return &plugin.Table{
 		Name:        "hibp_password",
 		Description: "Password (hashes) tracked by HIBP",
 		List: &plugin.ListConfig{
-			KeyColumns: plugin.AnyColumn([]string{"prefix", "hash"}),
+			KeyColumns: plugin.AnyColumn([]string{"plain", "hash"}),
 			Hydrate:    listPasswords,
 		},
-		Get: &plugin.GetConfig{
```
Comment (cbruno10) [inline hibp/table_password.go:20]: Are we not able to get a specific password hash anymore?

File: hibp/table_password.go [hunk 2]
```diff
@@ -2,125 +2,70 @@ package hibp
...
-			KeyColumns: plugin.SingleColumn("hash"),
-			Hydrate:    getPassword,
-		},
 		Columns: []*plugin.Column{
-			{Name: "prefix", Type: proto.ColumnType_STRING, Description: "The first five characters of the hash", Transform: transform.From(transformPrefix), Hydrate: hydratePrefix},
-			{Name: "hash", Type: proto.ColumnType_STRING, Description: "The hash of the compromised password.", Transform: transform.From(transformHash), Hydrate: hydrateHash},
+			{Name: "plain", Type: proto.ColumnType_STRING, Description: "The plain-text of the compromised password (sent as a hash to the API)."},
```
Comment (cbruno10) [inline hibp/table_password.go:27]: How does this change affect usage of the table? Am I still able to pass a prefix anywhere of 5 chars to search a range, like the API allows?

**Review by cbruno10 — COMMENTED** [review]
@binaek Please see comments, thanks!

File: hibp/table_password.go (hunk 2 above)
Comment (binaek) [reply hibp/table_password.go:27]: The API allows it, but the SDK does not expose all the results. I have opened up an issue at https://github.com/wneessen/go-hibp/issues/14 requesting for it.

File: config/hibp.spc [hunk 3]
```diff
@@ -1,4 +1,8 @@
 connection "hibp" {
-  plugin = "wedtm/hibp"
-  api_key = "replace-me-with-your-own-key"
-}
\ No newline at end of file
+  plugin  = "hibp"
+  
+  # Requests to HIBP API needs to carry an API KEY.
+  # You can get one at https://haveibeenpwned.com/API/Key
+  # These can also be 'HIBP_API_KEY'
```
Comment (cbruno10) [inline config/hibp.spc:6]: This sentence feels incomplete, can you please use the same wording from https://github.com/turbot/steampipe-plugin-mongodbatlas/blob/main/config/mongodbatlas.spc#L7-L9?

File: config/hibp.spc [hunk 4]
```diff
@@ -1,4 +1,8 @@
 connection "hibp" {
-  plugin = "wedtm/hibp"
-  api_key = "replace-me-with-your-own-key"
-}
\ No newline at end of file
+  plugin  = "hibp"
+  
+  # Requests to HIBP API needs to carry an API KEY.
```
Comment (cbruno10) [inline config/hibp.spc:4]:
```suggestion
  # Requests to HIBP API needs to carry an API key.
```

File: docs/index.md [hunk 5]
```diff
@@ -1,20 +1,20 @@
...
-description: Steampipe plugin to query Have I Been Pwned breaches, pastes, and passwords
-og_description: Query compromised data with SQL! Open source CLI. No DB required.
-og_image: "TBD"
+description: Steampipe plugin to query breaches, account breaches, pastes and passwords from Have I Been Pwned.
+og_description: Query HIBP data with SQL! Open source CLI. No DB required.
+og_image: "/images/plugins/turbot/hibp-social-graphic.png"
 ---
 
-# HaveIBeenPwned? + Steampipe
+# Have I Been Pwned? + Steampipe
```
Comment (cbruno10) [inline docs/index.md:13]:
```suggestion
# Have I Been Pwned + Steampipe
```

Not sure if we should include the `?` or not in titles here, README, etc., but we primarily need to be consistent, so if other places don't include `?`, we can leave it out here too

**Review by cbruno10 — COMMENTED** [review]
@binaek Added a few more suggestions I missed last time

Also, can you please add a `.github` directory (can copy the one from the AWS plugin) as well? Thanks!

File: hibp/client.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+		return "", fmt.Errorf("api key must be configured")
+	}
+
+	return apiKey, nil
+}
+
+func createClient(ctx context.Context, apiKey string) *hibp.Client {
+	clientOptions := []hibp.Option{
+		hibp.WithApiKey(apiKey),
+		hibp.WithUserAgent("Turbot Steampipe (+https://steampipe.io)"),
```
[the remark this answers is not in the document]
Comment (binaek) [reply hibp/client.go:57]: We typically don't set the user agent, but HIBP API doc says that it's one of their "best practices" that we should follow.

https://haveibeenpwned.com/API/v3#UserAgent

[the remark this answers is not in the document]
Comment (binaek) [reply hibp/client.go:57]: The format is adapted from `cli` (less the `version`)

https://github.com/turbot/steampipe/blob/main/utils/useragent.go#L17

File: docs/tables/hibp_password.md [hunk 7]
```diff
@@ -6,16 +6,24 @@ You must include at _least_ 5 characters in the `prefix` field.
```
Comment (cbruno10) [inline docs/tables/hibp_password.md:6]: Please update description above based on new table behaviour

File: docs/tables/hibp_password.md [hunk 8]
```diff
@@ -6,16 +6,24 @@ You must include at _least_ 5 characters in the `prefix` field.
...
 ## Examples
 
-### Show the number of times this password has been compromised
+### List the number of times a password hash has been compromised
 
 ```sql
-select * from hibp_password
- where hash = '5BAA61E4C9B93F3F0682250B6CF8331B7EE68FD8'
+select
+  *
```
Comment (cbruno10) [inline docs/tables/hibp_password.md:13]: Can you please replace `*` with either all named columns, or the important ones?

File: docs/tables/hibp_account.md [hunk 9]
```diff
@@ -1,26 +1,53 @@
 # Table: hibp_account
 
-This table returns data similar to the `hibp_breach` table, with the requirement of an `account` field.
+The most common use of the API is to return a list of all breaches a particular account has been involved in. This table returns data similar to the `hibp_breach` table, with the requirement and addition of an `account` field.
 
-While the `hibp_breaches` table will return all of the known breaches, there are millions of accounts associated with those
-breaches. To find the breaches associated with a certain account, you must use this table.
+While the `hibp_breaches` table will return all of the known breaches, this table can be used to find breaches for a particular account.
 
 ## Examples
```
Comment (cbruno10) [inline docs/tables/hibp_account.md:7]: Can you please add 1-2 examples of showing how to join this table with another plugin's tables on email addresses? For instance, getting the email from the `ldap_user` table and then showing breached accounts, and another example getting users from `okta_user` as well.

File: hibp/plugin.go [hunk 10]
```diff
@@ -16,10 +16,10 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 		},
 		DefaultTransform: transform.FromGo().NullIfZero(),
 		TableMap: map[string]*plugin.Table{
+			"hibp_account":  tableAccount(),
```
Comment (cbruno10) [inline hibp/plugin.go:19]:
```suggestion
			"hibp_breached_account":  tableBreachedAccount(),
```

I think breached account is a better table name, it's more descriptive and matches the API more closely.

**Review by cbruno10 — COMMENTED** [review]
Please see comments, thanks!

File: docs/tables/hibp_breached_account.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+### List breaches for active Okta users (requires [Okta plugin](https://hub.steampipe.io/plugins/turbot/okta))
+
+```sql
+select
+  title,
+  pwn_count,
+  breach_date
+from
+  hibp_breached_account,
```
Comment (cbruno10) [inline docs/tables/hibp_breached_account.md:63]:
```suggestion
  hibp_breached_account
```

This query will not run due to this syntax error, please test all queries before adding them to example docs.

File: docs/tables/hibp_breached_account.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+### List breaches for LDAP users (requires [LDAP plugin](https://hub.steampipe.io/plugins/turbot/ldap))
+
+```sql
+select
+  title,
+  pwn_count,
+  breach_date
+from
+  hibp_breached_account,
```
Comment (cbruno10) [inline docs/tables/hibp_breached_account.md:84]:
```suggestion
  hibp_breached_account
```

Similar to above, this query will not run.

**Review by cbruno10 — COMMENTED** [review]
@binaek Please check queries before adding them as examples, thanks!

**Pull request merged**